### PR TITLE
remove sub classes inheriting params

### PR DESCRIPTION
**Pull request by jlambert121**
Making changes without tests makes me nervous - next stuff will be tests.

Remove sub classes inheriting params
Fully qualify class inclusions
Move package resource to install class

### CODE REVIEW

**Comment by edestecd** [comment]
Why do we need to subscribe to the packages?
Most good package writers will restart the service on update...

Have you had a case of needing this?

**Comment by edestecd** [comment]
I noticed you changed $config_options in the clamd and freshclam classes did you update the var name in the erb templates?

**Comment by jlambert121** [comment]
Most good package writers will restart the service - but not all.  This just makes sure that it happens.

Templates updated as well now.

File: manifests/install.pp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# install.pp
+# Installs clamav package
+#
+
+class clamav::install {
+
+  package { 'clamav':
+    ensure  => installed,
+    name    => $clamav_package,
```
Comment (edestecd) [inline manifests/install.pp:11]: This variable needs to be fully qualified or have a local copy like the other classes...

File: manifests/install.pp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    require => Anchor['clamav::begin'],
```
Comment (edestecd) [inline manifests/install.pp]: Maybe we should drop this require, since the anchor is in the main class and not here.

File: manifests/install.pp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  }
+
+}
```
Comment (edestecd) [inline manifests/install.pp]: No newline at end of file?

**Comment by edestecd** [comment]
OK, the subscribe to package thing is fine by me.  Other than the comments I just made, everything else looks great.  Thanks for your work.  I can merge as soon as you fix the `clamav::install` var.

**Pull request merged**